src/chainparams.cpp

```diff
@@ -139,6 +139,9 @@ class CMainParams : public CChainParams {
         consensus.vUpgrades[Consensus::UPGRADE_NU5].nActivationHeight = 1687104;
         consensus.vUpgrades[Consensus::UPGRADE_NU5].hashActivationBlock =
             uint256S("0000000000d723156d9b65ffcf4984da7a19675ed7e2f06d9e5d5188af087bf8");
+        consensus.vUpgrades[Consensus::UPGRADE_NU6].nProtocolVersion = 0x7FFFFFFF;
```

Comment:
For consistency of the NU process, i think whatever we put here should be the same everywhere (including zebra).

I have a suggestion to use `0xffff_fffe` as the dummy value that can be interpreted as "the NU before the zfuture one". Unfortunately, the rust code is using `0xffff_ffff` for the zfuture NU while zcashd is using `0x7FFFFFFF` but we can change in one place if we want.

https://github.com/zcash/librustzcash/pull/1454/files#r1690364962

Nothing stops the different implementations to use its own values here. At the end, all we care is that the NU6 value has to be the same everywhere before activation but i think it might be good to use the same dummy values in preparation for that.

Reply:
I think that we should specify `0x7fffffff` as the ZFuture consensus branch ID in a ZIP, and go with that, since the zcashd value has the longest history. I don't know why ZFuture was set to `0xffffffff` in librustzcash; I have opened https://github.com/zcash/librustzcash/pull/1480 to bring these into alignment.

Reply:
> the zcashd value has the longest history.

In terms of time that a consensus branch ID was present in the `master` / `main` branch of the repo (which is what matters the most), this is false. I checked the history in https://github.com/zcash/librustzcash/pull/1480#discussion_r1700663788

I also no longer believe my hypothesis from today's meeting, because the consensus branch ID is not represented with a signed integer in `zcashd`:
https://github.com/zcash/zcash/blob/0bcddb896c51f58771a20966a0b1f5e5110a7e52/src/consensus/upgrades.h#L18-L20